src/Diag/Common.hs

```diff
@@ -0,0 +1,22 @@
+module Diag.Common (
+  MissingDeepDeps (..),
+  MissingEdges (..),
+  MissingClassificationBetweenDirectAndDeepDeps (..),
+) where
+
+import Diag.Diagnostic (ToDiagnostic (renderDiagnostic))
+
+data MissingDeepDeps = MissingDeepDeps
+instance ToDiagnostic MissingDeepDeps where
+  renderDiagnostic (MissingDeepDeps) =
+    "Could not analyze deep dependencies."
+
+data MissingEdges = MissingEdges
+instance ToDiagnostic MissingEdges where
+  renderDiagnostic (MissingEdges) =
+    "Could not analyze edges between dependencies."
+
+data MissingClassificationBetweenDirectAndDeepDeps = MissingClassificationBetweenDirectAndDeepDeps
```

Comment:
[nit]: Let's avoid massive names like this.  How about `AllDirectDeps`/`MissingRootEdge`?  The message can stay the same, but code-wise, this is pretty awkward.